File: src/classes_program2.rs
use std::env;
use std::fs::File;
use std::io::{self, BufRead, BufReader, Write};

fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() != 3 {
        eprintln!("Usage: {} <infile> <outfile>", args[0]);
        std::process::exit(1);
    }
    let infile_name = &args[1];
    let outfile_name = &args[2];
    let infile = File::open(infile_name).unwrap_or_else(|_| {
        eprintln!("Oops! Could not read {}", infile_name);
        std::process::exit(1);
    });
    let reader = BufReader::new(infile);
    let mut outfile = File::create(outfile_name).unwrap_or_else(|_| {
        eprintln!("Oops! Could not open {} for writing", outfile_name);
        std::process::exit(1);
    });
    for line in reader.lines() {
        match line {
            Ok(content) => {
                writeln!(outfile, "{}", content).unwrap();
            }
            Err(e) => {
                eprintln!("Error reading line: {}", e);
            }
        }
    }
    println!("Copied contents from {} to {}", infile_name, outfile_name);
}

File: python/classes_program2.py
import sys

if len(sys.argv) != 3:
    print(f"Usage: {sys.argv[0]} <infile> <outfile>")
    sys.exit(1)

infile_name = sys.argv[1]
outfile_name = sys.argv[2]

try:
    with open(infile_name, 'r') as infile:
        try:
            with open(outfile_name, 'w') as outfile:
                for line in infile:
                    outfile.write(line)
        except IOError:
            print(f"Oops! Could not open {outfile_name} for writing")
            sys.exit(1)
except IOError:
    print(f"Oops! Could not read {infile_name}")
    sys.exit(1)

print(f"Copied contents from {infile_name} to {outfile_name}")